macros/core/PGML.pl

```diff
@@ -1589,20 +1589,31 @@ sub Quote {
 }
 
 sub Rule {
-	my $self  = shift;
-	my $item  = shift;
-	my $width = " width:100%; ";
-	my $size  = "";
-	$width = ' width:' . $item->{width} . '; ' if defined $item->{width};
-	$size  = ' size="' . $item->{size} . '"'   if defined $item->{size};
-	my $html = '<hr' . $size . ' style="margin:.3em auto" />';
-	$html = '<div>'
-		. '<span style="'
-		. $width
-		. 'display:-moz-inline-box; display:inline-block; margin:.3em auto">'
-		. $html
-		. '</span>'
-		. '</div>';    # if $width ne '' && $item->{width} !~ m/%/;
+	my $self   = shift;
+	my $item   = shift;
+	my $width  = '100%;';
+	my $height = '1px';
+	if (defined $item->{width}) {
+		$width = $item->{width};
+		$width .= 'px' if ($width =~ /^\d*\.?\d+$/);
+	}
+	if (defined $item->{height}) {
+		$height = $item->{height};
+		$height .= 'px' if ($height =~ /^\d*\.?\d+$/);
+	} elsif (defined $item->{size}) {
+		$height = $item->{size};
+		$height .= 'px' if ($height =~ /^\d*\.?\d+$/);
+	}
+	my $html = main::tag(
+		'div',
+		main::tag(
+			'span',
+			style => "width:$width; display:inline-block; margin:0.3em auto",
+			main::tag(
+				'hr', style => "width:$width; height:$height; background-color:currentColor; margin:0.3em auto;"
+			)
+		)
+	);
 	return $self->nl . $html . "\n";
```

Comment:
Since the `$html` variable is not subsequently modified, there is no need for it.  This could just be:
```suggestion
	return $self->nl
		. main::tag(
			'div',
			main::tag(
				'span',
				style => "width:$width; display:inline-block; margin:0.3em auto",
				main::tag(
					'hr', style => "width:$width; height:$height; background-color:currentColor; margin:0.3em auto;"
				)
			)
		);
```
Note that I removed the `\n` at the end of that.  Historically there has been an attempt to make the generated html human readable.  That is bad practice for modern html.  Instead you should strive for minimized html which decreases download size and renders faster in the browser.  You can always use tools like prettier to make the html nicely formatted for human readability.  I have been removing these newlines from code whenever I make changes to related code.  (Unfortunately Mojolicious templates do not generate anything close to minimized html for webwork2.)